internal/commands/push/command.go

```diff
@@ -100,16 +100,17 @@ func (cmd *Command) Handler(profile *user.Profile, ui terminal.UI, clients cli.C
 		return err
 	}
 
-	var uploadPathDependencies string
+	var uploadDependenciesPath string
 	var dependenciesDiffs realm.DependenciesDiff
 	if cmd.inputs.IncludeDependencies {
-		uploadPathDependencies, err = local.PrepareDependencies(app, ui)
+		uploadDependencies, err := local.FindAppDependencies(app.RootDir)
 		if err != nil {
 			return err
 		}
-		defer os.Remove(uploadPathDependencies) //nolint:errcheck
+		uploadDependenciesPath := uploadDependencies.ArchivePath
```

Comment:
i actually think there's a subtle bug here... you can see it easier [here](https://play.golang.org/p/INJtTpTqdIg) but in short, b/c you have `:=` here you are no longer assigning to the string `var` we declared outside this if block